torchdata/dataloader2/communication/eventloop.py

```diff
@@ -35,24 +35,59 @@
 ]
 
 
+class _ResetCounter:
+    exp_cnt: int
+    cnt: int
+    _reached: bool
+
+    def __init__(self, exp_cnt: int):
+        self.exp_cnt = exp_cnt
+        self.cnt = 0
+        self._reached = False
+
+    def increment(self) -> None:
+        self.cnt += 1
+        assert self.cnt <= self.exp_cnt
+
+    def is_reached(self) -> bool:
+        if self._reached:
+            return self._reached
```

Comment:
nit: These two lines can be removed? But I guess it is slightly faster, so I'm indifferent.

Reply:
Lol, you are right. I will remove them tmrw